connectors/sources/confluence.py

```diff
@@ -222,17 +222,20 @@ async def api_call(self, url):
             response: Client response
         """
         self._logger.debug(f"Making a GET call for url: {url}")
-        try:
-            async with self._get_session().get(
-                url=url,
-                ssl=self.ssl_ctx,
-            ) as response:
-                yield response
-        except ServerDisconnectedError:
-            await self.close_session()
-            raise
-        except ClientResponseError as exception:
-            await self._handle_client_errors(url=url, exception=exception)
+        
+        while True:
```

Comment:
Ideally, it's not an infinite loop, because we have break and raise both conditions. So, after raising, the function will retry 3 times using the retryable and, at last the function will raise if an error is still present.

Reply:
But the caller function is retrying 3 times already, I don't really see how this change fixes the test. I don't think we should wrap one specific call in `while True` - this will fix the test potentially, but it's not the code that can go to production

Reply:
correct, this is a workaround. I am trying to change the approach and update it soon.